src/Action/Order/Assign.php

```diff
@@ -35,6 +47,42 @@ public function __invoke($data)
             $data->setCustomer($user->getCustomer());
         }
 
+        $body = [];
+        $content = $request->getContent();
+        if (!empty($content)) {
+            $body = json_decode($content, true);
+        }
+
+        if (isset($body['guest']) && true === $body['guest']) {
```

Comment:
I guess this is already an issue, but maybe we should check if the order is in an expected state here? The state should be `cart`.